### PR TITLE
Add t1-extensive pipeline

### LINKED ISSUE #31: Implement t1-extensive preprocessing pipeline
Extensive preprocessing (t1-volume-tissue-segmentation + skull-stripping) should be implemented.
Preprocessing subparser would have two options:
- `minimal` --> wrapper on clinica or message to the user.
- `extensive` --> perform extensive preprocessing.

### PR DESCRIPTION
**Pull request by alexandreroutier**
Closes #31

This PR adds the `t1-extensive`. I am asking review:

- from @14thibea to test this pipeline on 1 or 2 subjects from her paper (I only used `ub-ADNI002S0295_ses-M00` for comparison)
- from @mdiazmel for technical review
- from @nburgos for documentation review

Best,
Alex

### CODE REVIEW

File: docs/Preprocessing/T1_Extensive.md [hunk 1]
```diff
@@ -1 +1,25 @@
-# `t1-extensive` - Coming soon
+# `t1-extensive` - Skull-stripping of T1w images in MNI standard space
+
+This pipeline performs skull-stripping of T1-weighted MR images in [`Ixi549Space` space](https://bids-specification.readthedocs.io/en/stable/99-appendices/08-coordinate-systems.html) using brainmask from the [SPM](http://www.fil.ion.ucl.ac.uk/spm/) software
```
Comment (mdiazmel) [inline docs/Preprocessing/T1_Extensive.md]:
```suggestion
This pipeline performs skull-stripping of T1-weighted MR images in [`Ixi549Space` space](https://bids-specification.readthedocs.io/en/stable/99-appendices/08-coordinate-systems.html) using brainmask from the [SPM](http://www.fil.ion.ucl.ac.uk/spm/) software.
```

File: docs/Preprocessing/T1_Extensive.md [hunk 2]
```diff
@@ -1 +1,25 @@
-# `t1-extensive` - Coming soon
+# `t1-extensive` - Skull-stripping of T1w images in MNI standard space
```
Comment (nburgos) [inline docs/Preprocessing/T1_Extensive.md]: T1**-weighted MR** images (to be consistent with Clinica | BTW the t1-linear page of Clinica should be updated)

File: docs/Preprocessing/T1_Extensive.md (hunk 1 above)
Comment (nburgos) [inline docs/Preprocessing/T1_Extensive.md]: using the brain mask (`mask_ICV.nii`) from the [SPM](http://www.fil.ion.ucl.ac.uk/spm/) software.

File: docs/Preprocessing/T1_Extensive.md [hunk 3]
```diff
@@ -1 +1,25 @@
...
+
+This pipeline performs skull-stripping of T1-weighted MR images in [`Ixi549Space` space](https://bids-specification.readthedocs.io/en/stable/99-appendices/08-coordinate-systems.html) using brainmask from the [SPM](http://www.fil.ion.ucl.ac.uk/spm/) software
+
+
+## Prerequisites
+You need to execute the [`t1-volume` pipeline](http://www.clinica.run/doc/Pipelines/T1_Volume/) to run the pipeline. On the `t1-volume-tissue-segmentation` sub-pipeline ("Tissue segmentation, bias correction and spatial normalization to MNI space" step) needs to be executed.
```
Comment (nburgos) [inline docs/Preprocessing/T1_Extensive.md]: from Clinica to run this pipeline. Only the

**Review by mdiazmel — COMMENTED** [review]
It's ok for me. I tested with two subjects from Oasis without problems. 
Just have to solve a conflict with this file (`clinicadl/clinicadl/quality_check/quality_check.py`), probably due to a PR previously merged into `dev`.

File: clinicadl/clinicadl/preprocessing/t1_extensive/t1_extensive_cli.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    def define_options(self):
+        """Define the sub-command arguments."""
+        from clinica.engine.cmdparser import PIPELINE_CATEGORIES
+
+        # Clinica compulsory arguments (e.g. BIDS, CAPS, group_label)
+        clinica_comp = self._args.add_argument_group(
+            PIPELINE_CATEGORIES["CLINICA_COMPULSORY"]
+        )
+        clinica_comp.add_argument("caps_directory", help="Path to the CAPS directory.")
```
Comment (14thibea) [inline clinicadl/clinicadl/preprocessing/t1_extensive/t1_extensive_cli.py]: could be nice to have a more complete help line:
"Path to the CAPS directory containing the outputs of t1-volume."

**Review by 14thibea — COMMENTED** [review]
I tested on two subjects of ADNI and everything works well!

**Comment by alexandreroutier** [comment]
> It's ok for me. I tested with two subjects from Oasis without problems.
> Just have to solve a conflict with this file (`clinicadl/clinicadl/quality_check/quality_check.py`), probably due to a PR previously merged into `dev`.

Indeed, I push-forced my branch in order to fix this conflict.

Reviews from everyone were integrated in the last commits. I'll let @mdiazmel do the rebased merge if this is ok.

**Pull request merged**